src/analysis/vector/geometry_checker/qgsgeometrygapcheck.cpp

```diff
@@ -166,13 +166,15 @@ void QgsGeometryGapCheck::collectErrors( const QMap<QString, QgsFeaturePool *> &
     // Get neighboring polygons
     QMap<QString, QgsFeatureIds> neighboringIds;
     const QgsGeometryCheckerUtils::LayerFeatures layerFeatures( featurePools, featureIds.keys(), gapAreaBBox, compatibleGeometryTypes(), mContext );
+    std::unique_ptr< QgsGeometryEngine > gapGeomEngine = QgsGeometryCheckerUtils::createGeomEngine( gapGeom, mContext->tolerance );
+    gapGeomEngine->prepareGeometry();
```

Comment:
There's no need to prepare here -- the GEOS distance call doesn't take advantage of prepared geometries. @strk would it be possible to get a "GEOSPreparedDistance_r" call similar to your new "GEOSPreparedNearestPoints_r" one?